File: leetcode1371/xorPrefix.cpp
// class Solution 
// {
// public:
//     int findTheLongestSubstring(string s) 
//     {
//         const int vowelsBase=1065233,n=s.length();
        
//         int i,t,prefix=0,ans=0;
//         unordered_map<int,int>hash{{0,-1}};

//         for(i=0;i<n;++i)
//         {
//             prefix^=1<<(s[i]-'a');
            
//             t=prefix&vowelsBase;
//             if(hash.count(t))
//                 ans=max(ans,i-hash[t]);
//             else hash[t]=i;
//         }

//         return ans;
//     }
// };

File: leetcode1371/xorPrefix.go

// func findTheLongestSubstring(s string) int {
//     const vowelsBase = 1065233

// 	prefix := 0

// 	ans := 0

// 	hash := map[int]int{0: -1}

// 	for i, char := range s {

// 		char = unicode.ToLower(char)

// 		if strings.ContainsRune("aeiou", char) {

// 			prefix ^= 1 << (char - 'a')

// 		}

// 		t := prefix & vowelsBase

// 		if pos, ok := hash[t]; ok {

// 			ans = max(ans, i-pos)

// 		} else {

// 			hash[t] = i

// 		}

// 	}

// 	return ans

// }
